backend/group.py

```diff
@@ -0,0 +1,57 @@
+from flask import Response, request
+from flask_restful import Resource
+from database.models import Role, Groups
+from user import User
+from errors import PermissionError, NotATeacherError, NotAStudentError
+from mongoengine.errors import DoesNotExist
+from flask_jwt_extended import jwt_required, get_jwt_identity
+
+
+class Group(Resource):
+    def get(self, group_id):
+        try:
+            return Groups.objects.get(id=group_id)
+        except (DoesNotExist):
+            raise DoesNotExist
+
+    @jwt_required()
+    def put(self, group_id):
+        admin_id = get_jwt_identity()
+        if not User.isAdmin(self, admin_id):
+            raise PermissionError

```

Comment:
```
 if not User.isAdmin(self, admin_id):
            raise PermissionError
```
These lines are duplicated several times, move to a seperate method.

Reply:
fixed